src/libs/ajax/loaded-data/useLoadedData.ts

```diff
@@ -45,40 +58,45 @@ export type UseLoadedDataResult<T> = [LoadedState<T, unknown>, (dataCall: () =>
  */
 export const useLoadedData = <T>(hookArgs?: UseLoadedDataArgs<T>): UseLoadedDataResult<T> => {
   const args: UseLoadedDataArgs<T> = hookArgs || {};
-  const { onError } = args;
+  const { onError, onSuccess } = args;
   const [loadedData, setLoadedData] = useState<LoadedState<T, unknown>>({ status: 'None' });
   const previousStatus = usePrevious(loadedData.status);
   useEffect(() => {
     if (loadedData.status === 'Error' && previousStatus !== 'Error') {
       onError?.(loadedData);
     }
   }, [loadedData, previousStatus, onError]);
-  const updateDataFn = useCallback(async (dataCall: () => Promise<T>) => {
-    setLoadedData((previousLoadedData) => {
-      const previousState = previousLoadedData.status !== 'None' ? previousLoadedData.state : null;
-      return {
-        status: 'Loading',
-        state: previousState,
-      };
-    });
-    try {
-      const result = await dataCall();
-      setLoadedData({
-        status: 'Ready',
-        state: result,
-      });
-    } catch (err: unknown) {
-      const error = err instanceof Response ? Error(await err.text()) : err;
+  const updateDataFn = useCallback(
+    async (dataCall: () => Promise<T>) => {
       setLoadedData((previousLoadedData) => {
         const previousState = previousLoadedData.status !== 'None' ? previousLoadedData.state : null;
-        const errorResult: ErrorState<T, unknown> = {
-          status: 'Error',
+        return {
+          status: 'Loading',
           state: previousState,
-          error,
         };
-        return errorResult;
       });
-    }
-  }, []);
+      try {
+        const result = await dataCall();
+        const readyState: ReadyState<T> = {
+          status: 'Ready',
+          state: result,
+        };
+        setLoadedData(readyState);
+        onSuccess?.(readyState);
+      } catch (err: unknown) {
+        const error = err instanceof Response ? Error(await err.text()) : err;
+        setLoadedData((previousLoadedData) => {
+          const previousState = previousLoadedData.status !== 'None' ? previousLoadedData.state : null;
+          const errorResult: ErrorState<T, unknown> = {
+            status: 'Error',
+            state: previousState,
+            error,
+          };
+          return errorResult;
+        });
+      }
+    },
+    [onSuccess]
```

Comment:
hmmm... this was in response to a lint error since static analysis saw that onSuccess usage was now within scope of the updateData function.... but I can try the same useEffect breakout pattern you refer to...

Reply:
fixed